### DISCUSSION TITLE
Errors in choice() and .or()

### DISCUSSION THREAD

**Discussion by artegoser** [zesterer/chumsky#659] [Q&A] [answered]
Hi. I have multiple parsers, they are all in choice(). One of them gets an invalid argument error, but the main parser with choice() ends up getting a "multiple errors" error. Also, it identifies the correct span where the error occurred, but displays "multiple errors". How can i get the output of invalid argument errors.

I have read that in this case chumsky tries to get the most informative error, but for some reason it does not work for me.

Errors occured here:
```rust
let hlarg = num_arg
    .try_map(|n, s| Hl::try_from(n).map_err(|e| Rich::custom(s, e)));
```

Parsing of tag with this parser:

```rust
let h = just(Token::H)
    .ignore_then(hlarg)
    .then(text_body)
    .map(|(level, body)| H(body, level));
```

Main parser:

```rust
choice((
    el, h, p, br, ul, ol, row, link, navlink, btn, navbtn, img, table, tcol, tpcol, hr, b,
    i, bq, footlnk, footn, a, s, sup, sub, disc,
))
.or(choice((block, carousel, code, pre, meta)))
.or(choice((el_text, el_tags, paragraph)))
```

Source: https://github.com/TxtDot/dalet-rs/blob/main/src/daleth/parser/mod.rs

**Comment by zesterer** [comment, marked as answer]
When you say 'displays multiple errors', what do you mean? Are you seeing [this](https://docs.rs/chumsky/1.0.0-alpha.7/chumsky/error/enum.RichReason.html#variant.Many) variant?

**Reply by artegoser** [reply]
Oh, now I see it.  I was using ariadne and just calling the .to_string method on the message. This method was giving "multiple errors". I thought all errors were output separately and only the main parser output could have multiple errors. Anyway, now it makes sense. Thanks.

Perhaps for simplicity, the .to_string() method for Multiple variant can be changed so that it outputs the body of several errors at once, instead of writing "multiple errors".

**Reply by zesterer** [reply]
The `Many` variant is due for removal anyway. It is confusing to most users as you say, and has questionable benefits.
